django/utils/http.py

```diff
@@ -23,12 +23,12 @@
 ''', re.X)
 
 MONTHS = 'jan feb mar apr may jun jul aug sep oct nov dec'.split()
-__D = r'(?P<day>\d{2})'
-__D2 = r'(?P<day>[ \d]\d)'
+__D = r'(?P<day>[0-9]{2})'
+__D2 = r'(?P<day>[ 0-9][0-9])'
```

Comment:
Do we want to capture bad values early in the regex or leave it until later when constructing a `datetime` object as we currently do?

```suggestion
__D = r'(?P<day>0[1-9]|[12][0-9]|3[01])'
__D2 = r'(?P<day> [1-9]|[12][0-9]|3[01])'
```

Is is valid for `__D2`, i.e. `ASCTIME_DATE`, to contain a zero-padded day? (Given it's looking for space-padded.)